tests/test_m2m.py

```diff
@@ -0,0 +1,159 @@
+import unittest
+from m2m import m2m
+
+class MetadataRecordTests(unittest.TestCase):
+
+    def testMetadataRecordSetup(self):
+        # test that the creation of a record creates a MetadataRecord object.
+
+        self.record = m2m.MetadataRecord("mphillips")
+        self.assertIsInstance(self.record, m2m.MetadataRecord)
+
+    def testNoneElementValueEqualsNone(self):
+        # if element value is None then it should return None.
+
+        self.record = m2m.MetadataRecord("mphillips")
+        self.assertEquals(self.record.map("basic", "title", None), None)
+
+    def testEmptyElementValueEqualsNone(self):
+        # if element value is empty for non-required elements 
+        # then map should return None.
+
+        self.record = m2m.MetadataRecord("mphillips")
+        self.assertEquals(self.record.map("basic", "title", "", required=False), None)
+
+    def testSetBaseDirectory(self):
+
+        self.record = m2m.MetadataRecord("mphillips")
+        self.record.setBaseDirectory("out")
+        self.assertEquals(self.record.baseDirectory, "out")
+
+    def testSetFolderName(self):
+
+        self.record = m2m.MetadataRecord("mphillips")
+        self.record.setFolderName("folder")
+        self.assertEquals(self.record.foldername, "folder")
+
+    def testElementNotInFieldType(self):
+
+        with self.assertRaises(m2m.MetadataConverterException) as cm:
+            self.record = m2m.MetadataRecord("mphillips")
+            self.record.map("basic", "author", "text")
+
+        the_exception = cm.exception
+        self.assertEqual(the_exception.value, 'Element named "author" not in fieldTypes')
+
+    def testUnsupportedMappingFunctionType(self):
+
+        with self.assertRaises(m2m.MetadataConverterException) as cm:
+            self.record = m2m.MetadataRecord("mphillips")
+            self.record.map("simple", "title", "text")
+
+        the_exception = cm.exception
+        self.assertEqual(the_exception.value, 'Unsupported mapping function type, simple')
```

Comment:
I believe we can consolidate these two lines to 

```python
self.assertEqual(str(cm), 'Unsupported mapping function type, simple')
```
`str(e)` is the standard way of accessing the exception message in Python.